docs/NNNN-mutable-head.md

```diff
@@ -0,0 +1,439 @@
+object has not been modified between the time the mutable HEAD was created and
+it was committed.
+
+The mutable HEAD MUST be a valid OCFL version. Unlike normal versions, it MUST
+be stored at `[object-root]/extensions/NNNN-mutable-head/head`.
+
+The mutable HEAD inventory file MUST NOT be written to the object root, and the
+root inventory file MUST NOT reference files within the extension directory.
+
+If there is not an active mutable HEAD, then the extension directory MUST NOT
+exist. An object has an active mutable HEAD when a mutable HEAD inventory exists
+at `[object-root]/extensions/NNNN-mutable-head/head/inventory.json`. When there
+is an active mutable HEAD, new root object versions MUST NOT be created as this
+results in version conflicts between the root object and the mutable HEAD.
+
+The `revisions` directory MUST contain a revision marker for every mutation that
+has been made to the mutable HEAD. Revisions are discussed in more detail in the
+[revisions section](#revisions).
+
+#### Inventory
+
+When a mutable HEAD is created, if the HEAD version defined in the root
+inventory file is `N`, then the mutable HEAD version MUST be `N+1`. Regardless
+of how many times a mutable HEAD is mutated, its version MUST NEVER change. The
+mutable HEAD’s inventory file contains everything in the root inventory file,
+plus all content (manifest entries, version state, etc) pertinent to version
+`N+1`.
+
+Files that are added in the mutable HEAD are added to the inventory manifest as
+normal. Content paths MUST be relative to the object root. The manifest MUST NOT
+reference files in the mutable HEAD content directory that are no longer
+referenced in the mutable HEAD version’s state.
+
+The version fields `created`, `message`, and `user` SHOULD be overwritten on
+every update of the mutable HEAD.
+
+The following is an example mutable HEAD inventory file:
+
+```json
+{
+  "digestAlgorithm": "sha512",
+  "head": "v2",
+  "id": "ark:/12345/bcd987",
+  "manifest": {
+    "4d27c8...b53": [ "extensions/NNNN-mutable-head/head/content/r1/foo/bar.xml" ],
+    "9bb43j...n3a": [ "extensions/NNNN-mutable-head/head/content/r2/file1.txt" ],
+    "u8b99v...7b2": [ "extensions/NNNN-mutable-head/head/content/r3/file1.txt" ],
+    "7dcc35...c31": [ "v1/content/foo/bar.xml" ],
+    "cf83e1...a3e": [ "v1/content/empty.txt" ],
+    "ffccf6...62e": [ "v1/content/image.tiff" ]
+  },
+  "type": "https://ocfl.io/1.0/spec/#inventory",
+  "versions": {
+    "v1": {
+      "created": "2018-01-01T01:01:01Z",
+      "message": "Initial import",
+      "state": {
+        "7dcc35...c31": [ "foo/bar.xml" ],
+        "cf83e1...a3e": [ "empty.txt" ],
+        "ffccf6...62e": [ "image.tiff" ]
+      },
+      "user": {
+        "address": "alice@example.com",
+        "name": "Alice"
+      }
+    },
+    "v2": {
+      "created": "2018-02-02T02:02:02Z",
+      "message": "Fix bar.xml, remove image.tiff, add empty2.txt",
```

Comment:
For clarity, it may be helpful to add to the `message`: "..., rename file1.txt to file2.txt, add updated file1.txt"